### PR TITLE
Create and apply environments changes in concourse

### PR DESCRIPTION
**Pull request by lukaszraczylo**
- Determine which environment(s) definitions have changed in last git commit
- Apply changes to the desired environment(s)

Please note that concourse is currently down, therefore it's a blind commit and **should not be merged** yet

### CODE REVIEW

**Comment by lukaszraczylo** [comment]
Closes ministryofjustice/cloud-platform#38

**Comment by lukaszraczylo** [comment]
Status so far: https://concourse.apps.cloud-platform-live-0.k8s.integration.dsd.io/teams/main/pipelines/build-environments/jobs/prepare-environments-build/builds/15

File: pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+resources:
+- name: cloud-platform-environments-repo
+  type: git
+  source:
+    uri: https://github.com/ministryofjustice/cloud-platform-environments.git
+    branch: build-env-using-concourse
```
[the remark this answers is not in the document]
Comment (lukaszraczylo) [reply pipeline.yaml]: It is for development only, to show that 'it works'. It is going to be removed when everyone is happy with the work :)

[the remark this answers is not in the document]
Comment (lukaszraczylo) [reply pipeline.yaml]: @kalbir this one has been cleaned up

**Comment by lukaszraczylo** [comment]
@kalbir , @alkar , @razvan-moj ready for merge.

File: whichNamespace.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# This script determines which environment / cluster should we deploy to
+LIST_OF_CHANGED_FILES=`git diff-tree --no-commit-id --name-only -r $(git log --pretty=format:"%h" -1) | grep namespaces | cut -d "/" -f2 | uniq`
+# LIST_OF_CHANGED_FILES=`git diff-tree --no-commit-id --name-only -r afed721 | grep namespaces | cut -d "/" -f2 | uniq`
```
Comment (alkar) [inline whichNamespace.sh]: Cleanup please!

File: pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+resources:
+- name: cloud-platform-environments-repo
+  type: git
+  source:
+    uri: https://github.com/ministryofjustice/cloud-platform-environments.git
+  webhook_token: ((github-hook-id))
```
[the remark this answers is not in the document]
Comment (lukaszraczylo) [reply pipeline.yaml]: It is not managed yet - it's outside of the ticket scope at the moment from what I understood from last week's discussions.

File: whichNamespace.sh (hunk 2 above)
Comment (lukaszraczylo) [reply whichNamespace.sh]: Cleaned up.

**Comment by lukaszraczylo** [comment]
@alkar secrets reside in encrypted k8s config files now. Please review.

**Pull request merged**